js/main.js

```diff
@@ -0,0 +1,86 @@
+import  {addTask}  from "./addTask.js";
+import { postRequest } from "./postRequest.js";
+import { updateTaskCount } from "./updateTaskCount.js";
+import { saveToLocalStorage } from "./saveToLocalStorage.js";
+
+// Fetching TODO list from API
+async function fetchTodos() {
+    try{
+        const response = await fetch('https://dummyjson.com/todos');
+        if(!response.ok){
+            throw new Error(`Failed to create TODO, Status: ${response.status}`);
+        }
+        const data = await response.json();
+        displayTodos(data.todos);
+        localStorage.setItem("todos", JSON.stringify(data.todos));
+    }catch(error){
+        console.error('Error Fetching To do Data:', error.message);
+    }
+    
+}
+
+// Function to display TODO list
+function displayTodos(todos) {
+    const todoTableBody = document.getElementById('todo-table-body');
+    todoTableBody.innerHTML = ''; // Clear existing rows
+    todos.forEach(todo => addTask(todo, todoTableBody));
+}
+
+function loadTodos() {
+    const storedTodos = localStorage.getItem("todos");
+    if (storedTodos) {
+        displayTodos(JSON.parse(storedTodos));
```

Comment:
If the stored todos are invalid, `JSON.parse(todos)` will crash, so let's use try-catch to avoid `JSON.parse` errors

```
    try {
        const storedTodos = localStorage.getItem("todos");
        if (storedTodos) {
            displayTodos(JSON.parse(storedTodos));
        } else {
            fetchTodos();
        }
    } catch (error) {
        console.error("Error loading todos from local storage:", error.message);
        fetchTodos();
    }
```